### PR TITLE
Add support to skip tests that are not present in testrun (#50)

### PR DESCRIPTION
**Pull request by zeburek**

### CODE REVIEW

**Comment by apallier** [comment]
@zeburek Please, could you add a specific unittest of your feature?

**Comment by zeburek** [comment]
@apallier I'll try to make my best. But It could be hard for me, I don't have such an experience.

File: README.md [hunk 1]
```diff
@@ -110,4 +110,5 @@ py.test --testrail --tr-config=<settings file>.cfg
   --tr-dont-publish-blocked
                         Do not publish results of "blocked" testcases in
                         TestRail
+  --tr-skip-missing     Skip cases that not present in testrun
```
Comment (allankp) [inline README.md]: `Skip test cases that are not present in testrun`

File: README.rst [hunk 2]
```diff
@@ -116,6 +116,8 @@ All available options
       --tr-dont-publish-blocked
                             Do not publish results of "blocked" testcases in
                             TestRail
+      --tr-skip-missing     Skip cases that not present in testrun
```
Comment (allankp) [inline README.rst]: Just need to regen this to apply above change

File: pytest_testrail/conftest.py [hunk 3]
```diff
@@ -85,6 +85,11 @@ def pytest_addoption(parser):
         action='store_true',
         required=False,
         help='Determine if results of "blocked" testcases (in TestRail) are published or not')
+    group.addoption(
+        '--tr-skip-missing',
+        action='store_true',
+        required=False,
+        help='Skip cases that not present in testrun')
```
Comment (allankp) [inline pytest_testrail/conftest.py]: `Skip test cases that are not present in testrun`

File: pytest_testrail/plugin.py [hunk 4]
```diff
@@ -101,21 +101,24 @@ def clean_test_ids(test_ids):
...
+                    )
                 )
             )
     return testcaseids
 
 
 class PyTestRailPlugin(object):
     def __init__(self, client, assign_user_id, project_id, suite_id, cert_check, tr_name, run_id=0, plan_id=0,
-                 version='', close_on_complete=False, publish_blocked=True):
+                 version='', close_on_complete=False, publish_blocked=True, skip_missing=True):
```
Comment (allankp) [inline pytest_testrail/plugin.py]: Prefer this was set to False as a default as its an optional flag

File: tests/test_plugin.py [hunk 5]
```diff
@@ -91,7 +91,9 @@ def test_clean_test_ids():
 
 
 def test_get_testrail_keys(pytest_test_items, testdir):
-    assert plugin.get_testrail_keys(pytest_test_items) == [1234, 5678, 8765, 4321]
+    items = plugin.get_testrail_keys(pytest_test_items)
+    assert list(items[0][1]) == [1234, 5678]
+    assert list(items[1][1]) == [8765, 4321]
```
Comment (allankp) [inline tests/test_plugin.py]: Please add a test for your new feature.

**Comment by zeburek** [comment]
@allankp It seems that everything fixed
Unable to test same things in one test, so added two tests

**Comment by zeburek** [comment]
After merging this PR - I will open new one for #35 issue
My mistake with branches

**Comment by zeburek** [comment]
@allankp Would it be merged?

**Comment by allankp** [comment]
@zeburek sorry for the delay, thanks for your contribution

**Pull request merged**